src/dc_uudecode.c

```diff
@@ -32,31 +86,564 @@
+		ret_bytes = l * (-1);
+	}
+	else{
+		ret_bytes = 0;
+	}
+	
+	return ret_bytes;
+}
+
+
+char* dc_print_hex (char* s){
+	/**
+	 * make hex representation of a string
+	 * 
+	 * @Return
+	 *      hex representation of s (needs to be free'd)
+	 * 
+	 */
+	int n = strlen(s);
+	int i;
+	char* hex = (char*)calloc((2*n+1), sizeof(char));
+	
+	for(i = 0; i<n; i++){
+		sprintf(hex+i*2, "%02X", s[i]);
+	}
+	
+	return hex;
+	
+} //dc_print_hex()
+
+
+
+int dc_getline (char** ret_lineptr, char* source, char** ret_nextcharptr){
+	/**
+	 * 	Extracts first line from a given string (source)
+	 * 
+	 * 	Instead of strtok_r it doesn't manipulate the input string 
+	 *
+	 *  Parameters:
+	 * 
+	 *  @ret_lineptr   - returned line incl. line end char
+	 *  @source        - input string (will not be changed)
+	 *  @ret_nextchar  - pointer to next char after current line end
+	 *                   gives start ptr for next call of this function
+	 * 
+	 *  @Return: 
+	 *      strlen of line found (incl. line end char(s) but excluding '\0')
+	 *      0 if no line is read
+	 */
+	 
+	char* lineend   = source;
+	char* start     = source;
+	int   linelen   = 0;
+
+	*ret_lineptr = NULL;
+	
+	if((lineend = strstr(start, LF))){
+		
+		lineend++; // this belongs to the line
+		
+		linelen = lineend - start; // len of line
+
+		*ret_nextcharptr	= lineend;
+		
+		*ret_lineptr		= strndup(start, linelen);
+	}
+	
+	return linelen;
+} //dc_getline()
+
+
+
+int dc_uu_check_line (int n, int line_len){
+	/**
+	 *	Checks if n describes the correct line length for given encoded len char line len
+	 * 
+	 *  Parameters:
+	 * 
+	 *  @n        - number of encoded bytes in a line
+	 *  @line_len - number of chars in encoded string with first len char
+	 * 
+	 *  example uuencoded line where n is 45 and 1st char describes this:
+	 * 
+	 *     M9&%T``!3"F6(A '_F5V2'G']1(A5%)L-,KFI!G+OR#70]]L'X8H`?S S2I@P
+	 *     ||
+	 *     |+-- from here data
+	 *     +--- len char
+	 * 
+	 * 	@returns -	1 if line is ok
+	 * 				0 otherwise
+	 */
+	 
+	/* Calculate expected # of chars and pad if necessary */
+	int expected;
+	
+	if (n > 0){
+		expected = (n+2)/3*4;
+	}
+	else
+		return 0;
+
+	/* # of chars + len char is the expected line len*/
+	return (line_len == expected + 1) ? 1 : 0;
+	
+} //dc_uu_check_line()
+
+
+/**
+ *      Main uudecoding functions
+ */ 
+
+char* dc_find_uuencoded_part (const char* msgtxt){
+	/**
+	 *  Find uuencoded part in msgtxt & return it's position.
+	 *  If no part is found NULL is returned.
+	 */
+	 
+	char* uu_part_pos = NULL;
+	
+	char* tofind = "begin ";
+
+	if ((uu_part_pos = strstr(msgtxt, tofind))){
+		if (strncmp((uu_part_pos+9), " ", 1) == 0){ 
+			// second space after file rights checked, ("begin 666 filename")
+			// uuencoded part found
+			
+			return uu_part_pos;
+		}
+	}
+	
+	// here: no uuencoded text part identified
+	return NULL;
+	
+} //dc_find_uuencoded_part()
+
+
+char* dc_handle_uuencoded_part (const char*	msgtxt,
+								 char*			uu_msg_start_pos,
```

Comment:
nitpicking: tabs can only be used for indentation, for alignment, spaces are needed.
(maybe we should switch to spaces doe indentation as well, but this is a different discussion :)

Reply:
-> Doing update